tests/test_eigen.py

```diff
@@ -0,0 +1,144 @@
+import pytest
+try:
+    import numpy as np
+except ImportError:
+    np = None
+try:
+    import scipy
+except ImportError:
+    scipy = None
+from pybind11_tests import has_eigen
+
+requires_eigen_and_numpy = pytest.mark.skipif(not has_eigen or not np,
+                                              reason="eigen and/or numpy are not installed")
+requires_eigen_and_scipy = pytest.mark.skipif(not has_eigen or not scipy,
+                                              reason="eigen and/or scipy are not installed")
+
+ref = np.array([[ 0,  3,  0,  0,  0, 11],
+                [22,  0,  0,  0, 17, 11],
+                [ 7,  5,  0,  1,  0, 11],
+                [ 0,  0,  0,  0,  0, 11],
+                [ 0,  0, 14,  0,  8, 11]])
+
+
+def check(mat):
+    return np.sum(abs(mat - ref)) == 0
```

Comment:
It's much better to use `np.testing.assert_array_equal` than a home-baked solution (since it will report diffs between arrays)